shared/naturalcrit/splitPane/splitPane.jsx

```diff
@@ -1,200 +1,121 @@
 require('./splitPane.less');
 const React = require('react');
-const createClass = require('create-react-class');
+const { useState, useEffect, useRef } = React;
 const cx = require('classnames');
 
-const SplitPane = createClass({
-	displayName     : 'SplitPane',
-	getDefaultProps : function() {
-		return {
-			storageKey         : 'naturalcrit-pane-split',
-			onDragFinish       : function(){}, //fires when dragging
-			showDividerButtons : true
+const SplitPane = (props)=>{
+	const {
+		storageKey = 'naturalcrit-pane-split',
+		onDragFinish = ()=>{},
+		showDividerButtons = true
+	} = props;
+
+	const [isDragging, setIsDragging] = useState(false);
+	const [dividerPos, setDividerPos] = useState(null); // Initial divider position is set to `null`
+	const [moveSource, setMoveSource] = useState(false);
+	const [moveBrew, setMoveBrew] = useState(false);
+	const [showMoveArrows, setShowMoveArrows] = useState(true);
+	const [liveScroll, setLiveScroll] = useState(false);
+
+	const dividerRef = useRef(null);
+
+	// Set initial divider position and liveScroll only after mounting
+	useEffect(()=>{
+		const savedPos = window.localStorage.getItem(storageKey);
+		setDividerPos(savedPos ? parseInt(savedPos, 10) : window.innerWidth / 2);
+		setLiveScroll(window.localStorage.getItem('liveScroll') === 'true');
+
+		const handleResize = ()=>{
+			setDividerPos((pos)=>limitPosition(pos,0.1 * (window.innerWidth - 13), 0.9 * (window.innerWidth - 13))
+			);
 		};
-	},
-
-	getInitialState : function() {
-		return {
-			currentDividerPos : null,
-			windowWidth       : 0,
-			isDragging        : false,
-			moveSource        : false,
-			moveBrew          : false,
-			showMoveArrows    : true
-		};
-	},
-
-	pane1 : React.createRef(null),
-	pane2 : React.createRef(null),
-
-	componentDidMount : function() {
-		const dividerPos = window.localStorage.getItem(this.props.storageKey);
-		if(dividerPos){
-			this.setState({
-				currentDividerPos : this.limitPosition(dividerPos, 0.1*(window.innerWidth-13), 0.9*(window.innerWidth-13)),
-				userSetDividerPos : dividerPos,
-				windowWidth       : window.innerWidth
-			});
-		} else {
-			this.setState({
-				currentDividerPos : window.innerWidth / 2,
-				userSetDividerPos : window.innerWidth / 2
-			});
-		}
-		window.addEventListener('resize', this.handleWindowResize);
-
-		// This lives here instead of in the initial render because you cannot touch localStorage until the componant mounts.
-		const loadLiveScroll = window.localStorage.getItem('liveScroll') === 'true';
-		this.setState({ liveScroll: loadLiveScroll });
-	},
-
-	componentWillUnmount : function() {
-		window.removeEventListener('resize', this.handleWindowResize);
-	},
-
-	handleWindowResize : function() {
-		// Allow divider to increase in size to last user-set position
-		// Limit current position to between 10% and 90% of visible space
-		const newLoc = this.limitPosition(this.state.userSetDividerPos, 0.1*(window.innerWidth-13), 0.9*(window.innerWidth-13));
```

Comment:
There is one behavior difference I note. Maybe it's not critical but wanted to point it out. In the original behavior, changing the window size will remember the last divider position set by the user and *grow back to that size* if the window becomes larger again. In this PR, shrinking the window and making it large again does not regain the divider position.

First is the original behavior. Then I switch tabs to the new behavior.
https://github.com/user-attachments/assets/c4f7b6a7-f833-40af-86e9-4e65e5c57dbb

Reply:
Very interesting, i'll have a look

Reply:
```
const newLoc = this.limitPosition(this.state.userSetDividerPos, 0.1*(window.innerWidth-13), 0.9*(window.innerWidth-13));

this.setState({
			currentDividerPos : newLoc,
			windowWidth       : window.innerWidth
		});
```

Seems identical to 

`setDividerPos((pos)=>limitPosition(pos, 0.1 * (window.innerWidth - 13), 0.9 * (window.innerWidth - 13)));`

So i don't see where this could come from

Note: i erased the windowWidth state because it was not being referenced, just set. Not even in the original code.

Reply:
It's the `userSetDividerPos`, not `windowWidth`.

Reply:
Do you think this is a behavior worth holding on to or no?

Reply:
I think it is worth maintaining this behavior, if it was intentional.

Reply:
Fixed it without the extra state, which was causing issues:


https://github.com/user-attachments/assets/fac00044-4eb6-4df1-9e56-e5431adba140